bugsnag-android-performance/src/main/kotlin/com/bugsnag/android/performance/internal/ImmutableConfig.kt

```diff
@@ -37,12 +38,10 @@ internal data class ImmutableConfig(
         private const val VALID_API_KEY_LENGTH = 32
 
         private fun validateApiKey(apiKey: String) {
-            require(
-                apiKey.length == VALID_API_KEY_LENGTH
-                    && apiKey.all { it.isDigit() || it in 'a'..'f' },
+            if (apiKey.length != VALID_API_KEY_LENGTH
+                || !apiKey.all { it.isDigit() || it in 'a'..'f' }
             ) {
-
-                "Invalid configuration. apiKey should be a 32-character hexademical string, got '$apiKey' "
```

Comment:
We still want to throw an error and crash if the API key is missing, but log a warning (`Logger.w`) if it's invalid - that [matches the notifier behaviour I believe](https://github.com/bugsnag/bugsnag-android/blob/next/bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java#L52).

Reply:
The `apiKey` config is non-nullable (so there is a Kotlin intrinsic null-check in the setter). I've downgraded the log message to warning.

Reply:
No, Kotlin will throw an exception if you try set it to `null` in the config